### ISSUE TITLE
v2.1.2 added proprietary libraries

### ISSUE THREAD

**Issue by IzzySoft** [uintdev/qrserv#11]
Unfortunately, your adding the in-app update checker (which btw. Google frowns upon as well) added non-free dependencies to your otherwise fully foss app:

```
Offending libs:
---------------
* Google Mobile Services (/com/google/android/gms): NonFreeDep
* Google Play Core (/com/google/android/play/core): NonFreeNet,NonFreeDep

2 offenders.
```

So the mentioned 2 anti-features now apply. Would you consider removing that again (doesn't do much good anyway: the resp. store-apps notify about available updates themselves) – or at least have a FOSS flavor coming without those? Feels sad to have proprietary libraries in for a functionality one doesn't even need :wink:

Thanks in advance!

**Comment by uintdev** [comment]
I had a bit of a debate with this one. The intent was to inform of updates, as some seem to be a bit behind before reviewing. That's just the minority, though.
I wasn't aware that Google was against it, considering it does use their own APIs officially rather than being misused or utilising private APIs. So, that's a first I'm hearing of that.

I did notice that the application had been placed on FOSS repos a while back. So perhaps this was not the best move in this case. I'll get this one resolved shortly. Thanks for the pointer.

**Comment by uintdev** [comment]
I had now published a new release, v2.1.4, which addresses this. The release notes goes over some details and that affected releases have had notices added onto their own release notes, so that it's known what versions to avoid.

If there's anything else that needs sorting in this regard (i.e. ways of easing exclusion of affected releases), please do let me know. I'll leave this open for your response.

**Comment by IzzySoft** [comment]
> I wasn't aware that Google was against it

I don't have the full details, but I've read in multiple context they won't permit "auto-updaters" in apps. Maybe if it's just "update information" it would be permitted – but then, for those using Play the PS-app will probably have the update installed before your notification arrives. And for those not using Play, it's not much useful (except for the few "manual side-loaders" maybe).

> I did notice that the application had been placed on FOSS repos a while back.

Yes, it's e.g. in mine – so if you wish, be welcome to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link there e.g. from your Readme :wink: Ah, the "where": https://apt.izzysoft.de/packages/dev.uint.qrserv/

> So perhaps this was not the best move in this case. I'll get this one resolved shortly. Thanks for the pointer.

Gladly – and thanks for fixing it up that fast!

> I had now published a new release, v2.1.4, which addresses this.

Full success! So the anti-features never went live, awesome!

> If there's anything else that needs sorting in this regard

All looks fine, thanks again! One thing I can offer you is a PR with Fastlane metadata¹, so you'd be in control of descriptions & graphics as shown in my repo. See my [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) for some background (and for reference/guidance). If you want me to send what's configured here, just let me know and I open the corresponding PR ASAP.

Meanwhile, closing up here as this issue is solved. Your response concerning the PR will still reach me when done here, as I'll stay subscribed.

¹ <sub>no fastlane binaries needed for that concerning my repo – but if you wish you can of course use them to also use the very same metadata for Play Store</sub>
